### PR TITLE
Switch to retryableHttpClient for GitHub AuthN API Client + More Logs

### PR DESCRIPTION
**Pull request by yilmi**
Hi,

This pull request is meant to fix issue #994 and adds the following:
- Making GitHub authenticated API Client use the `retryableHttpClientTimeout()` by default
- Adding the repo URL to the log 
    - `"scanning git repo complete path: repoPath, time: scanTime"`
    - `"scanning git repo complete Repo: repoUrl path: repoPath, time: scanTime"`
- Adding a logger to all the httpClient supporting one
- Fixing minor documentation issue (missing flags)

This has already helped recover from requests failures as this dev build log output can show:
```json
{
    "level": "debug",
    "msg": "[DEBUG] GET https://api.github.com/repos/ORG/REPO: retrying in 1s (3 left)",
    "name": "RetryableHttpClientTimeout",
    "time": "2023-01-03T10:43:17Z"
}
{
    "level": "debug",
    "msg": "[DEBUG] GET https://api.github.com/orgs/ORG/repos?per_page=100 (status: 502): retrying in 1s (3 left)",
    "name": "RetryableHttpClientTimeout",
    "time": "2023-01-05T07:57:26Z"
}

{
    "level": "debug",
    "msg": "[DEBUG] GET https://keychecker.trufflesecurity.com/fingerprint/<FINGERPRINT> (status: 503): retrying in 0s (3 left)",
    "name": "RetryableHttpClient",
    "time": "2023-01-06T07:44:31Z"
}
```
This comes from the client - https://github.com/hashicorp/go-retryablehttp/blob/master/client.go#L690

-- EDIT: Added correct log output examples

### CODE REVIEW

**Review by dustin-decker — APPROVED** [review]
Thank you for the contribution!

File: README.md [hunk 1]
```diff
@@ -181,6 +182,7 @@ Flags:
       --no-verification          Don't verify the results.
       --only-verified            Only output verified results.
       --filter-unverified        Only output first unverified result per chunk per detector if there are more than one results.
+      --configFilename           Path to configuration file
```
Comment (dustin-decker) [inline README.md:185]: This flag should just be `--config`. You can get the output from `trufflehog --help` and paste it here.

Comment (yilmi) [reply README.md]: Thanks for the pointer, I updated this section with the latest CLI output when using `--help` 👍

**Review by dustin-decker — APPROVED** [review]
Thank you!

**Comment by bmac-io** [comment]
Withdrawing this report. Apologies for the noise.

**Pull request merged**